test/unit/compute_resource_test.rb

```diff
@@ -170,4 +170,43 @@ def setup
     cr = FactoryGirl.build(:compute_resource)
     refute as_admin { cr.send(:associate_by, 'mac', '11:22:33:44:55:1a') }.readonly?
   end
+
+  test "vm_compute_attributes_for works" do
+    fake_vm = OpenStruct.new
+    fake_vm.attributes = {:cpus => 2}
+    ComputeResource.any_instance.expects(:find_vm_by_uuid).returns(fake_vm)
+    cr = FactoryGirl.build(:compute_resource)
+    returned_vm = cr.vm_compute_attributes_for('test')
+    assert returned_vm[:cpus] == 2
```

Comment:
Style nitpick, prefer `assert_equal 2, returned_vm[:cpus]` as it provides better error messages when the assertion fails (showing the actual value).